More gracefully handle badging limitations (#123)
w3c/badging: explainer.md

The model does not allow a badge to be a negative integer, or the integer value 0 (setting the badge to 0 is equivalent to clearing the badge).

> **Important distinction**: A "flag" badge should display a visual indicator to the user (such as a dot or circle), whereas a cleared badge (value 0 or calling `clearAppBadge()`) should display nothing. These are semantically different states, and platforms must not treat a request to set a "flag" as equivalent to clearing the badge.

Author: yeah, will change to not use RFC language also in the explainer.

> **Important distinction**: A "flag" badge should display a visual indicator to the user (such as a dot or circle), whereas a cleared badge (value 0 or calling `clearAppBadge()`) should display nothing. These are semantically different states, and platforms must not treat a request to set a "flag" as equivalent to clearing the badge.

Reviewer: Just to make sure I get the difference between `"nothing"` and `0` for a value, is the former one just for initializing and upon runtime calls to clear the badge handle a value of `0`?

> **Important distinction**: A "flag" badge should display a visual indicator to the user (such as a dot or circle), whereas a cleared badge (value 0 or calling `clearAppBadge()`) should display nothing. These are semantically different states, and platforms must not treat a request to set a "flag" as equivalent to clearing the badge.

Author: They are equivalent. Setting it to `0` just sets it to "nothing".

> **Important distinction**: A "flag" badge should display a visual indicator to the user (such as a dot or circle), whereas a cleared badge (value 0 or calling `clearAppBadge()`) should display nothing. These are semantically different states, and platforms must not treat a request to set a "flag" as equivalent to clearing the badge.

Author: ```suggestion
**Important distinction**: A "flag" badge generally displays a visual indicator to the user (such as a dot or circle), whereas a cleared badge (value 0 or calling `clearAppBadge()`) clears it. These are semantically different states, and platforms generally won't treat a request to set a "flag" as equivalent to clearing the badge.
```

App badges are shown in OS-specific contexts. User agents should attempt reuse existing [operating system APIs and conventions](docs/implementation.md), to achieve a native look-and-feel for the badge.

### Implementation Considerations for Platform Limitations

> Different operating systems have varying support for badge types. Since the OS ultimately controls badge rendering, user agents must communicate the correct semantic intent:

Author: ```suggestion
Different operating systems have varying support for badge types. Since the OS ultimately controls badge rendering, it is the user agent that communicate the correct semantic intent:
```

> * **Flag support**: Some platforms do not natively support "flag" badges (badges without numbers). In such cases, user agents should communicate to the OS using the closest available representation that conveys badge presence (such as a generic symbol or the number "1") rather than requesting badge clearing.

Author: ```suggestion
* **Flag support**: Some platforms do not natively support "flag" badges (badges without numbers). In such cases, user agents communicate to the OS using the closest available representation that conveys badge presence (such as a generic symbol or the number "1") rather than requesting badge clearing.
```

> * **Number support**: Some platforms may not support numbered badges. In such cases, user agents may communicate the badge to the OS as a flag representation.

Author: ```suggestion
* **Number support**: Some platforms might not support numbered badges. In such cases, user agents can communicate the badge to the OS as a flag representation.
```

> * **Semantic preservation**: User agents must preserve the semantic distinction between "flag" (show something) and "nothing" (show nothing) when communicating with the operating system, even when platform capabilities are limited.

Author: ```suggestion
* **Semantic preservation**: User agents preserve the semantic distinction between "flag" (show something) and "nothing" (show nothing) when communicating with the operating system, even when platform capabilities are limited.
```

> * **OS control**: The ultimate display behavior is controlled by the operating system and may be further modified by user settings or system conventions.

Author: ```suggestion
* **OS control**: The ultimate display behavior is controlled by the operating system and can be further controlled by user settings or system conventions.
```
